tests/dat/actions/loggingTimeout.js

```diff
@@ -0,0 +1,35 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more contributor
+// license agreements; and to You under the Apache License, Version 2.0.
+
+// This action prints log lines to stdout for the duration of durationMillis.
+// The output is throttled by a delay of delayMillis between two log lines in
+// order to keep the log small and to stay within the log size limit.
+
+function doLog(startMillis, durationMillis, delayMillis) {
+  var lastLog = 0;
+  var logLines = 0;
+  while (true) {
+     now = new Date();
+     if (now - lastLog > delayMillis) {
+        lastLog = now;
+        logLines++;
+        console.log('[' + logLines + '] The quick brown fox jumps over the lazy dog.');
+     }
+     if (now - startMillis >= durationMillis) {
+        break;
+     }
+  }
+  return logLines;
+}
+
+
+function main(args) {
+   const durationMillis = args.durationMillis;
+   const delayMillis = args.delayMillis;
```

Comment:
Please use reasonable default values for `durationMillis` and `delayMillis` if these parameters were not passed. Or fail the action if these values are `undefined`.

Reply:
Done.